apps/api/src/app/step-schemas/usecases/get-step-schema/get-step-schema.usecase.ts

```diff
@@ -1,39 +1,149 @@
-import { Injectable } from '@nestjs/common';
-import {
-  ActionStepEnum,
-  actionStepSchemas,
-  ChannelStepEnum,
-  channelStepSchemas,
-  Schema,
-  StepType,
-} from '@novu/framework';
+import { BadRequestException, Injectable } from '@nestjs/common';
+
+import { JsonSchema, Schema, StepType } from '@novu/framework';
+import { NotificationStepEntity, NotificationTemplateRepository } from '@novu/dal';
 
-import { GetStepSchemaCommand } from './get-step-schema.command';
-import { SchemaOutput } from '../../types';
+import {
+  GetStepTypeSchemaCommand,
+  GetExistingStepSchemaCommand,
+  GetStepSchemaCommand,
+} from './get-step-schema.command';
+import { StepSchemaDto } from '../../dtos/step-schema.dto';
+import { mapStepTypeToOutput, mapStepTypeToResult } from '../../types';
 
 @Injectable()
 export class GetStepSchema {
-  async execute(command: GetStepSchemaCommand): Promise<SchemaOutput> {
-    return StepTypeToOutputMap[command.stepType];
+  constructor(private readonly notificationTemplateRepository: NotificationTemplateRepository) {}
+
+  async execute(command: GetStepSchemaCommand): Promise<StepSchemaDto> {
+    if (isGetByStepType(command)) {
+      return { controls: buildControlsSchema(command.stepType), variables: buildVariablesSchema() };
+    }
+
+    if (isGetByExistingStep(command)) {
+      const { currentStep, previousSteps } = await this.findSteps(command);
+
+      return {
+        controls: buildControlsSchema(currentStep?.template?.type as StepType),
+        variables: buildVariablesSchema(previousSteps),
+      };
+    }
+
+    throw new BadRequestException('Invalid command');
+  }
+
+  private async findSteps(command: GetExistingStepSchemaCommand) {
+    const workflow = await this.notificationTemplateRepository.findByIdQuery({
+      id: command.workflowId,
+      environmentId: command.environmentId,
+    });
+
+    if (!workflow) {
+      throw new BadRequestException(`No workflow found with the given id ${command.workflowId}`);
```

Comment:
Could we create a new error type to eliminate string literals in the code? This would provide a clearer separation of concerns and improve searchability of issues, as it would avoid parameterization in the error messages.

Reply:
i refactored it to a simple message with params.
regarding the new error type, lets make a decision in the team on what solution we want to go with and then do it all together, right now i will keep it as is because i am afraid we will end up with a code base that looks like 1K people maintaining.